pilot/pkg/networking/core/v1alpha3/listener_builder.go

```diff
@@ -311,26 +311,23 @@ func (lb *ListenerBuilder) patchListeners() {
 	}
 	lb.virtualOutboundListener = patchOneListener(lb.virtualOutboundListener, networking.EnvoyFilter_SIDECAR_OUTBOUND)
 	lb.virtualInboundListener = patchOneListener(lb.virtualInboundListener, networking.EnvoyFilter_SIDECAR_INBOUND)
-	lb.inboundListeners = envoyfilter.ApplyListenerPatches(networking.EnvoyFilter_SIDECAR_INBOUND, lb.node,
-		lb.push, lb.inboundListeners, false)
 	lb.outboundListeners = envoyfilter.ApplyListenerPatches(networking.EnvoyFilter_SIDECAR_OUTBOUND, lb.node,
 		lb.push, lb.outboundListeners, false)
 }
 
 func (lb *ListenerBuilder) getListeners() []*xdsapi.Listener {
 	if lb.node.Type == model.SidecarProxy {
-		nInbound, nOutbound := len(lb.inboundListeners), len(lb.outboundListeners)
+		nOutbound := len(lb.outboundListeners)
```

Comment:
I remember this file also had a boolean flag (reduceInboundListenersToFilterChains).. if it still exists, dont forget to nuke that as well.

Reply:
useInboundFilterChain seems to be removed by another pr